plaso/parsers/windefenderhistory.py

```diff
@@ -0,0 +1,429 @@
+# -*- coding: utf-8 -*-
+"""Parser for Windows Defender DetectionHistory files."""
+
+from datetime import datetime, timedelta
+import os
+
+from dtfabric.runtime import data_maps as dtfabric_data_maps
+from dfdatetime import time_elements as dfdatetime_time_elements
+
+from plaso.containers import events
+from plaso.containers import time_events
+
+from plaso.lib import definitions
+from plaso.lib import dtfabric_helper
+from plaso.lib import errors
+
+from plaso.parsers import interface
+from plaso.parsers import manager
+
+
+class WinDefenderHistoryEventData(events.EventData):
+  """Windows Defender Detection History event data.
+
+    Attributes:
+        sha256(str): Hash of the file.
+        filename(str): Filename that the threat was detected in.
+        extra(str): Comma separated list of extra detection locations.
+        container_filename(str): If the detected file was in a container,
+            its location.
+        web_filename(str): If the detected file was downloaded from the web,
+            its URI.
+        threatname(str): The threat that was detected.
+        host_and_user(str): Endpoint name in DOMAIN\\USER format.
+        process(str): The process that cased the detection.
+  """
+
+  DATA_TYPE = "av:windows:defenderlog"
+
+  def __init__(self):
+    """Initializes event data."""
+    super(WinDefenderHistoryEventData, self).__init__(
+        data_type=self.DATA_TYPE
+    )
+    self.sha256 = None
+    self.filename = None
+    self.extra = None
+    self.container_filename = None
+    self.web_filename = None
+    self.threatname = None
+    self.host_and_user = None
+    self.process = None
+
+
+class WinDefenderHistoryParser(
+    interface.FileObjectParser, dtfabric_helper.DtFabricHelper
+):
+  """Parses the Windows Defender History Log."""
+
+  NAME = "windefenderhistory"
+
+  _FILE_SIGNATURE = "Magic.Version"
+
+  _VALUE_DESCRIPTIONS = [
+      {0: "Threat identifier", 1: "Identifier"},
+      {0: "UnknownMagic1", 1: "Threat name", 4: "Category"},
+      {
+          0: "UnknownMagic2",
+          1: "Resource type",
+          2: "Resource location",
+          4: "Threat tracking data size",
+          5: "Threat tracking data",
+          6: "Last threat status change time",
+          12: "Domain user1",
+          14: "Process name",
+          18: "Initial detection time",
+          20: "Remediation time",
+          24: "Domain user2",
+      },
+  ]
+
+  _CATEGORY_NAME = {
```

Comment:
Let's move these to output formatting helpers instead

Reply:
These are not actually used anywhere so I will remove them and we can keep them in the dtFormats repo in case they're needed down the track